### PR TITLE
astropy units and tables

### PR DESCRIPTION
**Pull request by Dolgalad**
Solves Issue #44

### CODE REVIEW

**Comment by jeandet** [comment]
> This pull request **introduces 1 alert** when merging [e6b1d4e](https://github.com/SciQLop/speasy/commit/e6b1d4e0888c07383b3874fbabfed272f12ff934) into [6282950](https://github.com/SciQLop/speasy/commit/6282950d59d0079fe59698d95f971809e02160ec) - [view on LGTM.com](https://lgtm.com/projects/g/SciQLop/speasy/rev/pr-3cb961379d06527979f490b43d1d6f763e52999a)
> 
> **new alerts:**
> 
> * 1 for Except block handles 'BaseException'

I think this one is quite legit, you should catch a ValueError.

**Review by jeandet — CHANGES_REQUESTED** [review]
It looks good to me, except for the LGTM issue and it has 0 unit test. 
I think this must be tested with at leas one product of each provider and maybe with and without units to ensure we have a decent statement coverage and also that later we won't brake this.

**Comment by Dolgalad** [comment]
> It looks good to me, except for the LGTM issue and it has 0 unit test. I think this must be tested with at leas one product of

Tests for `to_astropy_table` implemented in `tests.test_spwc_variable`

File: speasy/products/variable.py [hunk 1]
```diff
@@ -115,6 +118,29 @@ def to_dataframe(self, datetime_index=False) -> pds.DataFrame:
             time = self.time
         return pds.DataFrame(index=time, data=self.values, columns=self.columns, copy=True)
 
+    def to_astropy_table(self, datetime_index=False) -> pds.DataFrame:
```
Comment (jeandet) [inline speasy/products/variable.py:121]:
```suggestion
    def to_astropy_table(self, datetime_index=False) -> astropy.table.Table:
```

File: speasy/products/variable.py [hunk 2]
```diff
@@ -115,6 +118,29 @@ def to_dataframe(self, datetime_index=False) -> pds.DataFrame:
...
+        ----------
+        datetime_index: bool
+            boolean indicating that the index is datetime
+
+        Returns
+        -------
+        astropy.Table:
+            Variable converted to astropy.Table
+        """
+        # try to get the units
```
Comment (jeandet) [inline speasy/products/variable.py:134]:
```suggestion
```
This comments brings nothing

File: tests/test_spwc_variable.py [hunk 3]
```diff
@@ -116,6 +119,19 @@ def setUp(self):
...
+        # valid astropy unit
+        var.meta = {"PARAMETER_UNITS": "nT"}
+        at = var.to_astropy_table()
+        self.assertTrue(isinstance(at, astropy.table.Table))
+        self.assertTrue(at["Values"].unit == astropy.units.nT)
+
+        # invalid astropy unit
+        var.meta["PARAMETER_UNITS"] = "not a unit"
+        at = var.to_astropy_table()
+        self.assertTrue(at["Values"].unit is None)
```
Comment (jeandet) [inline tests/test_spwc_variable.py:134]:
```suggestion
        self.assertTrue(isinstance(at, astropy.table.Table))
        self.assertTrue(at["Values"].unit is None)
```

**Review by jeandet — CHANGES_REQUESTED** [review]
Last few cosmetic changes and I merge.

**Pull request merged**